WordPress/gutenberg
# [RNMobile] Prevent calling the registerComponent callback multiple times

## Description


On development mode, I noticed that the callback executed when registering a component in the App registry is called multiple times. This case happens when the code within the callback triggers any warning or error (this means calling `console.warning` or `console.error`), in [this issue](https://github.com/WordPress/gutenberg/issues/32882#issuecomment-868414379) there's additional helpful information regarding this.

~~This issue has been addressed by adding a static object that contains the components that have been already registered, this way we enforce that the callback's code will be executed only once.~~

**EDIT:** I finally followed a different approach to address this problem, based on the [`AppRegistry` documentation](https://reactnative.dev/docs/appregistry), I created a class component that holds the initialization logic.

## How has this been tested?



1. Add the following code `debugger;` before [this line](https://github.com/WordPress/gutenberg/blob/dc27a49391947e2430010956a66afab4d95cc1d7/packages/element/src/react-platform.native.js#L25)
2. Run the app connected with Metro and debug it with Chrome
3. Open a post/page
4. Observe that the debugger gets paused due to the code above.
5. Resume the script execution <img width="34" alt="Screenshot 2021-06-25 at 13 35 20" src="https://user-images.githubusercontent.com/14905380/123419187-32eda300-d5ba-11eb-87ab-f3ce8929d32c.png">
6. Observe that the debugger is not paused again.

## Verify that reusable blocks are rendered

The original issue was spotted due to the reusable blocks being rendered as unsupported blocks so it would be nice to verify that the issue is no longer happening.

1. Open a post/page that contains reusable blocks (if required add them via the web version of the editor)
2. Observe that the reusable blocks are rendered properly

## Screenshots 
N/A

## Types of changes




Bug fix

## Checklist:
